tensorflow/python/data/experimental/ops/get_single_element.py

```diff
@@ -20,31 +20,118 @@
+  processed_features = tf.data.get_single_element(dataset)
+  ```
+
+  In the above example, the `raw_features` tensor of length=BATCH_SIZE
+  was converted to a `tf.data.Dataset`. Next, each of the `raw_feature` was
+  mapped using the `preprocessing_fn` and the processed features were
+  grouped into a single batch. The final `dataset` contains only one element
+  which is a batch of all the processed features.
+
+  NOTE: The `dataset` should contain only one element.
+
+  Now, instead of creating an iterator for the `dataset` and retrieving the
+  batch of features, the `tf.data.get_single_element()` function is used
+  to skip the iterator creation process and directly output the batch of
+  features.
+
+  This can be particularly useful when your tensor transformations are
+  expressed as `tf.data.Dataset` operations, and you want to use those
+  transformations while serving your model. Especially, when it comes to
+  serving `tf.keras` models using a `SavedModel`.
+
+  tf.keras
+  --------
+
+  ```python
+
+  model = ... # A pre-built or custom model
+
+  class PreprocessingModel(tf.keras.Model):
+    def __init__(self, model):
+      super().__init__(self)
+      self.model = model
+
+    @tf.function(input_signature=[
+      tf.TensorSpec([None, 28, 28],dtype=tf.uint8)
+    ])
+    def serving_fn(self, data):
+      # Strictly speaking, you do not need to use `tf.data` here at all, as simple
+      # `data / 255` would suffice. However, the point here is to illustrate that
+      # `tf.data` can be leveraged for efficient processing of data.
+      ds = tf.data.Dataset.from_tensor_slices(data)
+      ds = ds.map(lambda x: x / 255, num_parallel_calls=16)
+      ds = ds.batch(batch_size=16)
+      probabilities = self.model(tf.data.experimental.get_single_element(ds))
+      return tf.argmax(probabilities, axis=-1)
+
+  preprocessing_model = PreprocessingModel(model)
+  your_exported_model_dir = ... # save the model to this path.
+  tf.saved_model.save(preprocessing_model, your_exported_model_dir,
+                signatures={'serving_default': preprocessing_model.serving_fn})
+  ```
+
+  tf.estimator
+  ------------
+
+  In the case of estimators, you need to generally define a `serving_input_fn`
+  which would require the features to be processed by the model while
+  inferencing.
+
+  ```python
+  def serving_input_fn():
+
+    raw_feature_spec = ... # Spec for the raw_features
+    input_fn = tf.estimator.export.build_parsing_serving_input_receiver_fn(
+        raw_feature_spec, default_batch_size=None)
+    )
+    serving_input_receiver = input_fn()
+    raw_features = serving_input_receiver.features
+
+    def preprocessing_fn(raw_feature):
+      # ... the raw_feature is preprocessed as per the use-case
+      return feature
+
+    dataset = (tf.data.Dataset.from_tensor_slices(raw_features)
+              .map(preprocessing_fn, num_parallel_calls=BATCH_SIZE)
+              .batch(BATCH_SIZE))
```

Comment:
@jsimsa should we add some info on what the `BATCH_SIZE` can be? It should ideally be equal to the number of elements in `raw_features`. WDYT?

Reply:
Sounds good. Adding a comment that the code assume that the leading dimension of `raw_features` is `BATCH_SIZE` should be sufficient.

Reply:
done!